packages/sdk/src/oracle/store/index.ts

```diff
@@ -0,0 +1,18 @@
+import type { State } from "../types/state";
+import Write from "./write";
+import produce from "immer";
+
+export type WriteCallback = (write: Write) => void;
+export type Store = {
+  write: (cb: WriteCallback) => void;
+};
+
+export function factory(emit: (state: State, prev: State) => void = () => ({}), state: State = {}): Store {
```

Comment:
Could you add some natspec comments explaining how this is intended to be used? emit and state are both pretty general variable names, so it's unclear what they're meant to be set to.

Reply:
yes, ive refactored this into a class, and added comments